include/measurement_kit/nettest.hpp

```diff
@@ -0,0 +1,1922 @@
+}
+
+void NoisyRouter::on_status_measurement_start(const events::StatusMeasurementStartEvent &event) {
+    std::clog << "status.measurement_start";
+    std::clog << ":";
+    std::clog << " idx='" << event.idx << "'";
+    std::clog << " input='" << event.input << "'";
+    std::clog << std::endl;
+}
+
+void NoisyRouter::on_status_measurement_submission(const events::StatusMeasurementSubmissionEvent &event) {
+    std::clog << "status.measurement_submission";
+    std::clog << ":";
+    std::clog << " idx='" << event.idx << "'";
+    std::clog << std::endl;
+}
+
+void NoisyRouter::on_status_measurement_done(const events::StatusMeasurementDoneEvent &event) {
+    std::clog << "status.measurement_done";
+    std::clog << ":";
+    std::clog << " idx='" << event.idx << "'";
+    std::clog << std::endl;
+}
+
+void NoisyRouter::on_status_report_close(const events::StatusReportCloseEvent &event) {
+    std::clog << "status.report_close";
+    std::clog << ":";
+    std::clog << " report_id='" << event.report_id << "'";
+    std::clog << std::endl;
+}
+
+void NoisyRouter::on_status_report_create(const events::StatusReportCreateEvent &event) {
+    std::clog << "status.report_create";
+    std::clog << ":";
+    std::clog << " report_id='" << event.report_id << "'";
+    std::clog << std::endl;
+}
+
+void NoisyRouter::on_status_resolver_lookup(const events::StatusResolverLookupEvent &event) {
+    std::clog << "status.resolver_lookup";
+    std::clog << ":";
+    std::clog << " ip_address='" << event.ip_address << "'";
+    std::clog << std::endl;
+}
+
+void NoisyRouter::on_status_started(const events::StatusStartedEvent &event) {
+    std::clog << "status.started";
+    (void)event; /* No event attributes */
+    std::clog << std::endl;
+}
+
+void NoisyRouter::on_status_update_performance(const events::StatusUpdatePerformanceEvent &event) {
+    std::clog << "status.update.performance";
+    std::clog << ":";
+    std::clog << " direction='" << event.direction << "'";
+    std::clog << " elapsed='" << event.elapsed << "'";
+    std::clog << " num_streams='" << event.num_streams << "'";
+    std::clog << " speed_kbps='" << event.speed_kbps << "'";
+    std::clog << std::endl;
+}
+
+void NoisyRouter::on_status_update_websites(const events::StatusUpdateWebsitesEvent &event) {
+    std::clog << "status.update.websites";
+    std::clog << ":";
+    std::clog << " url='" << event.url << "'";
+    std::clog << " status='" << event.status << "'";
+    std::clog << std::endl;
+}
+
+void NoisyRouter::on_task_terminated(const events::TaskTerminatedEvent &event) {
+    std::clog << "task_terminated";
+    (void)event; /* No event attributes */
+    std::clog << std::endl;
+}
+
+NoisyRouter::~NoisyRouter() noexcept {}
+
+} // namespace routers
+
+// Nettest
+// -------
+
+Nettest::Nettest() noexcept {}
+
+Nettest::~Nettest() noexcept {}
+
+bool Nettest::start_captive_portal(settings::CaptivePortalSettings settings, routers::Router *router) noexcept {
```

Comment:
These `start_XXX` methods are auto-generated. (I need to map in more nettest specific settings. Until I do that, they will not appear in this implementation.)